toolchain/sem_ir/ids.h

```diff
@@ -52,6 +52,33 @@ struct InstId : public IdBase<InstId> {
 
 constexpr InstId InstId::InitTombstone = InstId(NoneIndex - 1);
 
+// And InstId whose value is a type. The fact it's a type is CHECKed on
+// construction, and this allows that check to be represented in the type
+// system.
+struct TypeInstId : public InstId {
+  explicit TypeInstId(File& sem_ir, InstId id);
+
+  // Ergonomic helper for `Check::Context`, which is not visible from SemIR.
+  template <class Context>
+  explicit TypeInstId(Context& context, InstId id)
+      : TypeInstId(context.sem_ir(), id) {}
+
```

Comment:
I think these are unused now. (And the first one seems to not be defined.)
```suggestion
```

Reply:
Oops, yes thanks.